index.ts

```diff
@@ -166,7 +168,7 @@ function getDarwinUrl(version: string, options: Options): string | null {
 
   const darwin = version === "9.0.0" ? "-darwin-apple" : "-apple-darwin";
   const prefix = "clang+llvm-";
-  const suffix = `-x86_64${darwin}${DARWIN_VERSIONS[version] ?? ""}.tar.xz`;
+  const suffix = `-${options.arch ?? "x86_64"}${darwin}${DARWIN_VERSIONS[version] ?? ""}.tar.xz`;
```

Comment:
I'm interested in this feature, too. However, there is one extra detail that I would need to make it work for our use case.

I would like to specify `arch: arm64`, but the darwin suffix versions are different. (`21.0` and `22.0`)
```
clang+llvm-15.0.7-arm64-apple-darwin22.0.tar.xz
clang+llvm-15.0.7-x86_64-apple-darwin21.0.tar.xz
```

To solve this I think it would work to key the `DARWIN_VERSIONS` by version and arch.
e.g.
```typescript
/** The Darwin version suffixes which are applied for some releases. */
const DARWIN_VERSIONS: { [key: string]: string } = {
  "15.0.7-x86_64": "21.0",
  "15.0.7-arm64": "22.0",
};
```
Then when building the `suffix` pass the version and arch concatenated.

Reply:
Well, ideally it'd just download the correct version for host architecture instead of us specifying all of these custom options